src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -222,6 +222,19 @@ public static IServiceCollection AddApplicationInsightsTelemetryProcessor(this I
             return services.AddSingleton<ITelemetryProcessorFactory>(serviceProvider => new TelemetryProcessorFactory(serviceProvider, telemetryProcessorType));
         }
 
+        /// <summary>
+        /// Extension method to provide configuration logic for application insights telemetry module.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> instance.</param>
+        /// <param name="configModule">Action used to configure the module.</param>
+        /// <returns>
+        /// The <see cref="IServiceCollection"/>.
+        /// </returns>        
+        public static IServiceCollection ConfigureTelemetryModule<T>(this IServiceCollection services, Action<T> configModule) where T : ITelemetryModule
+        {
+            return services.AddSingleton(typeof(ITelemetryModuleConfigurator), (new TelemetryModuleConfigurator(config => configModule((T)config), typeof(T))));
```

Comment:
Nit: Remove the unnecessary parentheses around the 2nd arg to AddSingleton.

Reply:
done